api/v1alpha1/oadp_types.go

```diff
@@ -54,47 +54,77 @@ const CSIPluginImageKey UnsupportedImageKey = "csiPluginImageFqin"
+	RestoreResourcesVersionPriority string `json:"restoreResourcesVersionPriority,omitempty"`
+	// If you need to install Velero without a default backup storage location NoDefaultBackupLocation flag is required for confirmation
 	// +optional
-	CustomVeleroPlugins []CustomPlugin `json:"customVeleroPlugins,omitempty"`
+	NoDefaultBackupLocation bool `json:"noDefaultBackupLocation,omitempty"`
+	// Pod specific configuration
+	PodConfig *PodConfig `json:"podConfig,omitempty"`
+}
 
-	// EnableRestic is a boolean to specify if restic daemonset instance should be created or not
+// PodConfig defines the pod configuration options
+type PodConfig struct {
+	// NodeSelector defines the nodeSelector to be supplied to Restic podSpec
 	// +optional
-	EnableRestic *bool `json:"enableRestic,omitempty"`
-	// ResticSupplementalGroups defines the linux groups to be applied to the Restic Pod
+	NodeSelector map[string]string `json:"nodeSelector,omitempty"`
+	// Tolerations defines the list of tolerations to be applied to Restic daemonset
 	// +optional
-	ResticSupplementalGroups []int64 `json:"resticSupplementalGroups,omitempty"`
-	// ResticNodeSelector defines the nodeSelector to be supplied to Restic podSpec
+	Tolerations []corev1.Toleration `json:"tolerations,omitempty"`
+	// ResourceAllocations defines the CPU and Memory resource allocations for the restic Pod
 	// +optional
-	ResticNodeSelector map[string]string `json:"resticNodeSelector,omitempty"`
-	// ResticTolerations defines the list of tolerations to be applied to Restic daemonset
+	ResourceAllocations corev1.ResourceRequirements `json:"resourceAllocations,omitempty"`
+}
+
+// ResticConfig is the configuration for restic server
+type ResticConfig struct {
+	// SupplementalGroups defines the linux groups to be applied to the Restic Pod
 	// +optional
-	ResticTolerations []corev1.Toleration `json:"resticTolerations,omitempty"`
-	// ResticResourceAllocations defines the CPU and Memory resource allocations for the restic Pod
+	SupplementalGroups []int64 `json:"supplementalGroups,omitempty"`
+	// Timeout defines the Restic timeout, default value is 1h
 	// +optional
-	ResticResourceAllocations corev1.ResourceRequirements `json:"resticResourceAllocations,omitempty"`
-	// ResticTimeout defines the Restic timeout, default value is 1h
+	Timeout string `json:"timeout,omitempty"`
+	// Pod specific configuration
+	PodConfig *PodConfig `json:"podConfig,omitempty"`
+}
+
+// ApplicationConfig defines the configuration for the Data Protection Application
+type ApplicationConfig struct {
+	Velero *VeleroConfig `json:"velero,omitempty"`
+	Restic *ResticConfig `json:"restic,omitempty"`
+}
+
+// BackupLocation defines the configuration for the DPA backup storage
+type BackupLocation struct {
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	Velero            *velero.BackupStorageLocationSpec `json:"velero"`
+}
+
+// VolumeSnapshot defines the configuration for the DPA snapshot store
+type VolumeSnapshot struct {
```

Comment:
Going with XLocation pattern. Should this be SnapshotLocation? DiskSnapshotLocation, StorageSnapshotLocation